Support for WebTestClient

Josef-Reichardt opened the pull request:
Closes #238.

fbenz commented on spring-auto-restdocs-core/pom.xml:
```diff
@@ -89,7 +94,6 @@
             <groupId>org.apache.tomcat.embed</groupId>
             <artifactId>tomcat-embed-el</artifactId>
             <version>9.0.2</version>
-            <scope>test</scope>
```
Is there a reason why this was changed? I think that the test scope is sufficient. We want to force as few dependencies on Spring Auto REST Docs clients as possible.

Josef-Reichardt replied:
Absolutely. Can't remember why I have removed the `test` scope. I think I have tried something... I have added it scope again now.

Josef-Reichardt commented:
Neither Mono nor Flux was supported correctly. I have adjusted the example application now to have one method which uses `Mono<...>` and one which uses `Flux<...>` as return type and realized that for both methods the `auto-response-fields` snippet was empty. To resolve this issue I have added the two types in `JacksonResponseFieldSnippet`. Now it looks good for Mono **and** Flux.

Josef-Reichardt commented on spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonResponseFieldSnippet.java:
```diff
@@ -71,6 +71,19 @@ protected Type getType(final HandlerMethod method) {
         } else {
+            try {
+                Class<?> monoClass = Class.forName("reactor.core.publisher.Mono");
+                Class<?> fluxClass = Class.forName("reactor.core.publisher.Flux");
```
Yes it's the same here. And yes it's really better/cleaner to do the type check this way and without the try/catch.
I've changed it and also added unit tests for the Mono and Flux return types.

Josef-Reichardt commented on samples/java-webtestclient/pom.xml:
```diff
@@ -0,0 +1,257 @@
+            <!--<groupId>org.springframework.security.oauth</groupId>-->
+            <!--<artifactId>spring-security-oauth2</artifactId>-->
+            <!--<version>2.3.3.RELEASE</version>-->
+        <!--</dependency>-->
```
Done :+1:

Josef-Reichardt commented on samples/java-webtestclient/generated-docs/index.html:
```diff
@@ -0,0 +1,2471 @@
+</div>
+<div class="sect3">
+<h4 id="_authorization_4"><a class="anchor" href="#_authorization_4"></a><a class="link" href="#_authorization_4">2.4.1. Authorization</a></h4>
+<div 
```
URL is missing in docs if URI is set by UriBuilder.

Example:
```
webTestClient.get()
    .uri(uriBuilder -> uriBuilder.path("/items/search")
        .queryParam("desc", "main")
        .queryParam("hint", "1")
        .build())
    // ...
```
I will look at this.

Josef-Reichardt replied:
Fixed: 9bf28b0

jmisur commented:
let me merge this already. if you find some improvements or bugfixes, just submit a PR. We should be much faster with additional smaller PRs then.
